### PR TITLE
Improve FIO benchmarks

### PR DESCRIPTION
**Pull request by cvlc**
This updates the benchmarks for v2.6 and as a first step extracts out some shared functions to make it easier to manage or add more going forward.

It uses trap to catch an exit or break and cleanup and the more conventional `env bash`

### CODE REVIEW

**Review by vfiftyfive — COMMENTED** [review]
Great job on these enhancements!

File: FIO/local-volumes/dbench-job-generator-local-volume-replica.sh [hunk 1]
```diff
@@ -15,78 +18,33 @@ set -euo pipefail
...
+echo -e "${GREEN}CLI pod: ${cli}"
 
 # Get the node name and id where the volume will get provisioned and attached on
 # Using the StorageOS cli is guarantee that the node is running StorageOS
-node_details=$(kubectl -n ${STOS_NS} exec cli -- storageos describe nodes -ojson | jq -r '[.[0].labels."kubernetes.io/hostname",.[0].id]')
-local_node_name=$(echo $node_details | jq -r '.[0]')
-local_node_id=$(echo $node_details | jq -r '.[1]')
+nodes_and_ids=($(get_storageos_nodes))
+node_name=${nodes_and_ids[0]#*~}
+node_id=${nodes_and_ids[0]%~*}
```
Comment (vfiftyfive) [inline FIO/local-volumes/dbench-job-generator-local-volume-replica.sh:34]: There is an inconsistency with these expansions, name and id are inverted when using the script with replica
```suggestion
node_name=$(echo ${nodes_and_ids[0]} | cut -d "~" -f1)
node_id=$(echo ${nodes_and_ids[0]} | cut -d "~" -f2)
```

File: FIO/local-volumes/dbench-job-generator-local-volume.sh [hunk 2]
```diff
@@ -15,80 +19,32 @@ set -euo pipefail
...
 # Get the node name and id where the volume will get provisioned and attached on
 # Using the StorageOS cli is guarantee that the node is running StorageOS
-node_details=$(kubectl -n ${STOS_NS} exec cli -- storageos describe nodes -o json | jq -r '[.[0].labels."kubernetes.io/hostname",.[0].id]')
-# The ID of the 1st node in the cluster
-node_id=$(echo $node_details | jq -r '.[1]')
-# The Name of the 1st node in the cluster
-node_name=$(echo $node_details | jq -r '.[0]')
+nodes_and_ids=($(get_storageos_nodes))
+node_name=${nodes_and_ids[0]%~*}
+node_id=${nodes_and_ids[0]#*~}
```
Comment (vfiftyfive) [inline FIO/local-volumes/dbench-job-generator-local-volume.sh:35]:
```suggestion
node_name=$(echo ${nodes_and_ids[0]} | cut -d "~" -f1)
node_id=$(echo ${nodes_and_ids[0]} | cut -d "~" -f2)
```

File: FIO/lib/sh/functions.sh [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+  pod=$(kubectl get pod -l job-name="${job}" --no-headers -ocustom-columns=_:.metadata.name 2>/dev/null || :)
+  SECONDS=0
+  TIMEOUT=420
+  while ! kubectl get pod "${pod}" -otemplate="{{ .status.phase }}" 2>/dev/null| grep -q Succeeded; do
+    pod_status=$(kubectl get pod "${pod}" -otemplate="{{ .status.phase }}" 2>/dev/null)
+    if [ $SECONDS -gt $TIMEOUT ]; then
+        echo "The pod $pod didn't succeed after $TIMEOUT seconds" 1>&2
+        echo -e "${GREEN}Pod: ${pod}, is in ${pod_status}${NC} state."
+        # Cleanup if job fails for any reason
+        kubectl delete -f "${manifest}"
```
[the remark this answers is not in the document]
Comment (cvlc) [reply FIO/lib/sh/functions.sh]: Thanks, great point!

**Review by vfiftyfive — APPROVED** [review]

**Pull request merged**